homeassistant/components/fritzbox/__init__.py

```diff
@@ -11,80 +12,103 @@
     CONF_USERNAME,
     EVENT_HOMEASSISTANT_STOP,
 )
-from homeassistant.helpers import discovery
 import homeassistant.helpers.config_validation as cv
 
-_LOGGER = logging.getLogger(__name__)
+from .const import CONF_CONNECTIONS, DEFAULT_HOST, DEFAULT_USERNAME, DOMAIN, PLATFORMS
 
-SUPPORTED_DOMAINS = ["binary_sensor", "climate", "switch", "sensor"]
 
-DOMAIN = "fritzbox"
-
-ATTR_STATE_BATTERY_LOW = "battery_low"
-ATTR_STATE_DEVICE_LOCKED = "device_locked"
-ATTR_STATE_HOLIDAY_MODE = "holiday_mode"
-ATTR_STATE_LOCKED = "locked"
-ATTR_STATE_SUMMER_MODE = "summer_mode"
-ATTR_STATE_WINDOW_OPEN = "window_open"
+def ensure_unique_hosts(value):
+    """Validate that all configs have a unique host."""
+    vol.Schema(vol.Unique("duplicate host entries found"))(
+        [socket.gethostbyname(entry[CONF_HOST]) for entry in value]
+    )
+    return value
 
 
 CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_DEVICES): vol.All(
-                    cv.ensure_list,
-                    [
-                        vol.Schema(
-                            {
-                                vol.Required(CONF_HOST): cv.string,
-                                vol.Required(CONF_PASSWORD): cv.string,
-                                vol.Required(CONF_USERNAME): cv.string,
-                            }
-                        )
-                    ],
-                )
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_DEVICES): vol.All(
+                        cv.ensure_list,
+                        [
+                            vol.Schema(
+                                {
+                                    vol.Required(
+                                        CONF_HOST, default=DEFAULT_HOST
+                                    ): cv.string,
+                                    vol.Required(CONF_PASSWORD): cv.string,
+                                    vol.Required(
+                                        CONF_USERNAME, default=DEFAULT_USERNAME
+                                    ): cv.string,
+                                }
+                            )
+                        ],
+                        ensure_unique_hosts,
+                    )
+                }
+            )
+        },
+        extra=vol.ALLOW_EXTRA,
+    )
 )
```

Comment:
```suggestion
    ),
    extra=vol.ALLOW_EXTRA,
)
```

Reply:
I can confirm, that this solves the issue